pkg/ccl/backupccl/restore_job.go

```diff
@@ -1459,21 +1459,10 @@ func (r *restoreResumer) doResume(ctx context.Context, execCtx interface{}) erro
 	defer func() {
 		mem.Shrink(ctx, memSize)
 	}()
-	// backupCodec is the codec that was used to encode the keys in the backup. It
-	// is the tenant in which the backup was taken.
-	backupCodec := keys.SystemSQLCodec
-	if len(sqlDescs) != 0 {
```

Comment:
i've removed this extra condition around checking if len(sqlDescs) !=0. I'm not really sure what it adds. I think it's to curb against decoding the tenant prefix for a system tenant backup, but I believe `latestBackupManifest.HasTenants()` would prevent decoding a span from a system tenant backup as well.